packages/manager/modules/pci/src/projects/project/notebooks/add/components/notebook-attach/notebook-attach.html

```diff
@@ -0,0 +1,165 @@
+<div>
+    <!--Notebook attach object storage-->
+    <div class="mb-4">
+        <!--Title-->
+        <p class="oui-heading_5 mb-2">
+            <strong data-translate="pci_notebook_add_attach_object"></strong>
+        </p>
+
+        <!--Description-->
+        <span>
+            <p
+                class="mb-0"
+                data-translate="pci_notebook_add_attach_container_description"
+            ></p>
+            <a data-ui-sref="pci.projects.project.storages" target="_blank">
+                <span
+                    data-translate="pci_notebook_add_attach_container_description_link"
+                ></span>
+            </a>
+        </span>
+    </div>
+
+    <!--Notebook storage labels-->
+    <div data-ng-if="$ctrl.display" class="mb-4">
+        <!--Title-->
+        <p class="oui-heading_5 mb-2">
+            <strong
+                data-translate="pci_notebook_add_configuration_label"
+            ></strong>
+        </p>
+
+        <!--Description-->
+        <p class="mb-1"
+            data-translate="pci_notebook_add_configuration_label_description"
+        ></p>
+
+        <!--Add label-->
+        <oui-button
+            class="mb-2"
+            icon-left="oui-icon-plus"
+            variant="secondary"
+            disabled="!$ctrl.canAttachVolumes()"
+            on-click="$ctrl.displayInlineAdder = true"
+        >
+            <span
+                data-translate="pci_notebook_add_attach_container_btn_attach_storage"
+            ></span>
+        </oui-button>
+
+        <!--Edit label-->
+        <oui-inline-adder
+            class="mb-1"
+            ng-if="$ctrl.displayInlineAdder"
+            on-add="$ctrl.onInlineAdderAddBtnClick(form)"
+            on-remove="$ctrl.onInlineAdderRemoveBtnClick(form)"
+        >
+            <oui-inline-adder-row class="pt-2" size="l"
+                  ng-init="storage = $ctrl.constructor.createStorage(null, null, null, null)">
```

Comment:
is it possible to avoid createStorage(null, null, null, null)? its not readable and difficult to understand. May be you can create a function like initializeStorage() or something

Reply:
This method create an object storage (empty).
i rename it to `createEmptyStorage()`.